### ISSUE TITLE
failure on assembly language

### ISSUE THREAD

**Issue by kt0755** [S2E/docs#25]
Hello
I'm working on concolic testing on linux kernel. In particular, extracting collected constraint information. But I think symbolic state can't go further on some assembly part ( e.g. copy_from_user). Is there any idea to continue to update symbolic state(or store)? I guess one naive solution is to transform it to C code.

**Comment by adrianherrera** [comment]
Hi @kt0755,

S2E executes binaries in QEMU, so there is absolutely no issue with it executing programs written in assembly. There is no need to transform it to C code - it would simply get compiled back into a binary anyway.

What exactly is the "failure"? Does S2E "get stuck"? There are many reasons for this. There is a current issue with the clocks that is causing symbolic execution to get stuck - see my comment [here](https://github.com/S2E/docs/issues/22#issuecomment-325593921) to see what branches you can pull to fix this issue. Otherwise, I recommend reading the [docs](https://github.com/S2E/docs/blob/master/src/FAQ.rst) for other ideas.

Sorry, without further details I cannot give any more help.

Regards,
Adrian

**Comment by kt0755** [comment]
Thank you for your reply. Let me make clear my question. (my failure is not "get stuck")
I just want to see the all path constraints along one path with the given input. 
Following is the short example.

s2e_disable_forking()
s2e_make_concolic(src, sizeof(src), "src") // like this
...
if (src == "A") XX;
...
copy_from_user(dst, src, size);  // there is asm inside this function.
...
if (dst == "B") XX;

I thought I expect to see both "A" and "B" as printed constraints (actually dumped during addConstraint()). But I only saw "A".  Then I just figured out "copy_from_user" function has inline asm code inside. That's why I think S2E cannot go further on asm code like "copy_from_user" case. 

Am I correct? I hope this is not stupid question. Thanks.
